HBFA/UefiHostTestPkg/Library/MemoryAllocationLibHost/MemoryAllocationLibHost.c

```diff
@@ -262,12 +262,7 @@ ReallocatePool (
   IN VOID   *OldBuffer  OPTIONAL
   )
 {
-  VOID  *NewBuffer;
-  NewBuffer = malloc (NewSize);
-  if (NewBuffer != NULL && OldBuffer != NULL) {
-    memcpy (NewBuffer, OldBuffer, MIN (OldSize, NewSize));
-  }
-  return NewBuffer;

```

Comment:
Would it be too annoying to change the function definition since now NewSize is not needed anymore? And is OldBuffer really optional anymore?

Reply:
OldBuffer is optional. If ptr is [NULL](https://en.cppreference.com/w/c/types/NULL), the behavior is the same as calling [malloc](http://en.cppreference.com/w/c/memory/malloc)(new_size). And NewSize is needed, OldSize isn't but I wouldn't touch the function definition cause who knows what that may break.